text-generation-inference/server/text_generation_server/cli.py

```diff
@@ -62,8 +61,33 @@ def serve(
 
     # Downgrade enum into str for easier management later on
     quantize = None if quantize is None else quantize.value
-    dtype = None if dtype is None else dtype.value
-    server.serve(model_id, revision, dtype, uds_path)
+    dtype = "bfloat16" if dtype is None else dtype.value
+
+    logger.info("CLI SHARDED = {} DTYPE = {}".format(sharded, dtype))
+
+    if sharded:
+        tgi_file =  None
+        for file in Path('/usr/local').rglob("tgi_service.py"):
+            tgi_file = file.absolute()
```

Comment:
I don't understand why we are doing this.
This `tgi_service.py` is the file that is in the same folder as `cli.py` right?

Reply:
yes but default running dir is /usr/src/ so we need to specify the absolute path. 
I can try setting the PYTHONPATH and see if that helps

Reply:
yeah we need to specify full path.

Reply:
Why not using something like that? https://github.com/huggingface/optimum-habana/blob/6c9f2198cfef3f1ced7426b52a1a15544aae9c7c/tests/test_text_generation_example.py#L58

Pathlib enables to get the absolute path to the current file, go up to a parent and then specify the name of the file.

Reply:
yeah this a good suggestion. I like this :-)